From 07ac1c11dc890fb8aaf2b9df4de490dc28d1b79b Mon Sep 17 00:00:00 2001
From: Misael Lopez Cruz <misael.lopez@ti.com>
Date: Fri, 10 Jun 2011 13:12:51 -0500
Subject: [PATCH 54/54] ASoC: ABE HAL: Fix CMEM address for write_gain

Fix CMEM offset used for omap_abe_write_gain(), the address must be
aligned to 32-bits.

Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
---
 sound/soc/omap/abe/abe_gain.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/omap/abe/abe_gain.c b/sound/soc/omap/abe/abe_gain.c
index a3d2b07..a31894d 100644
--- a/sound/soc/omap/abe/abe_gain.c
+++ b/sound/soc/omap/abe/abe_gain.c
@@ -645,14 +645,14 @@ int omap_abe_write_gain(struct omap_abe *abe,
 	/* CMEM bytes address */
 	mixer_target = OMAP_ABE_C_1_ALPHA_ADDR;
 	/* a pair of gains is updated once in the firmware */
-	mixer_target += (p + mixer_offset) << 1;
+	mixer_target += ((p + mixer_offset) >> 1) << 2;
 	/* load the ramp delay data */
 	omap_abe_mem_write(abe, OMAP_ABE_CMEM, mixer_target,
 		       (u32 *) &alpha, sizeof(alpha));
 	/* CMEM bytes address */
 	mixer_target = OMAP_ABE_C_ALPHA_ADDR;
 	/* a pair of gains is updated once in the firmware */
-	mixer_target += (p + mixer_offset) << 1;
+	mixer_target += ((p + mixer_offset) >> 1) << 2;
 	omap_abe_mem_write(abe, OMAP_ABE_CMEM, mixer_target,
 		       (u32 *) &beta, sizeof(beta));
 	return 0;
-- 
1.7.4.1

